File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int tar) {

     HashMap<Integer, Integer> map = new HashMap<>();  

     for(int i=0; i< nums.length; i++){
        int t = tar-nums[i];

        if(map.containsKey(t)){
            return new int[]{map.get(t), i};
        }
        map.put(nums[i], i);

     }

     return new int[]{-1};

    }
}

File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], tar: int) -> int:

        l = 0;
        r = len(nums) -1 

        while l <= r:
            mid = (l + r) //2

            if nums[mid] == tar:
                return mid
            elif nums[mid] > tar:
                r = mid - 1
            else:
                l = mid + 1
        return -1;                

        